receiver/hostmetricsreceiver/hostmetrics_receiver_test.go

```diff
@@ -55,30 +58,26 @@ func TestGatherMetrics_EndToEnd(t *testing.T) {
 	require.NoError(t, err, "Failed to start metrics receiver: %v", err)
 	defer func() { assert.NoError(t, receiver.Shutdown(context.Background())) }()
 
-	require.Eventually(t, func() bool {
-		got := sink.AllMetrics()
-		if len(got) == 0 {
-			return false
+	time.Sleep(180 * time.Millisecond)
+
+	got := sink.AllMetrics()
+
+	// expect 2 MetricData objects
+	assert.Equal(t, 2, len(got))
+
+	// extract the names of all returned metrics
+	metricNames := make(map[string]bool)
+	for _, metricData := range got {
+		metrics := internal.AssertMetricDataAndGetMetricsSlice(t, metricData)
+		for i := 0; i < metrics.Len(); i++ {
+			metricNames[metrics.At(i).MetricDescriptor().Name()] = true
 		}
+	}
 
-		assertMetricData(t, got)
-		return true
-	}, time.Second, 10*time.Millisecond, "No metrics were collected")
-}
+	// expect 2 metrics
+	assert.Equal(t, 2, len(metricNames))
 
-func assertMetricData(t *testing.T, got []pdata.Metrics) {
-	metrics := internal.AssertSingleMetricDataAndGetMetricsSlice(t, got)
-
-	// expect 1 metric
-	assert.Equal(t, 1, metrics.Len())
-
-	// for cpu seconds metric, expect a datapoint for each state label & core combination with at least 4 standard states
-	hostCPUTimeMetric := metrics.At(0)
-	internal.AssertDescriptorEqual(t, cpuscraper.MetricCPUSecondsDescriptor, hostCPUTimeMetric.MetricDescriptor())
-	assert.GreaterOrEqual(t, hostCPUTimeMetric.Int64DataPoints().Len(), runtime.NumCPU()*4)
-	internal.AssertInt64MetricLabelExists(t, hostCPUTimeMetric, 0, cpuscraper.CPULabel)
-	internal.AssertInt64MetricLabelHasValue(t, hostCPUTimeMetric, 0, cpuscraper.StateLabel, cpuscraper.UserStateLabelValue)
-	internal.AssertInt64MetricLabelHasValue(t, hostCPUTimeMetric, 1, cpuscraper.StateLabel, cpuscraper.SystemStateLabelValue)
-	internal.AssertInt64MetricLabelHasValue(t, hostCPUTimeMetric, 2, cpuscraper.StateLabel, cpuscraper.IdleStateLabelValue)
-	internal.AssertInt64MetricLabelHasValue(t, hostCPUTimeMetric, 3, cpuscraper.StateLabel, cpuscraper.InterruptStateLabelValue)
+	// expected metric names
+	assert.Contains(t, metricNames, "host/cpu/time")
+	assert.Contains(t, metricNames, "host/memory/used")
```

Comment:
I simplified the e2e test quite a lot so it now only validates that we get all of the metrics expected, but doesn't validate the content of these metrics (which can be validated by tests in the scrapers). This means we don't replicate nearly identical validation logic for the scrapers and receivers, and also means we don't need to export constant values from the scrapers just to be used by this test (in a following PR I'll unexport the cpu metric constants). But it does make the e2e test a bit less thorough.